src/v0/sources/pipedream/util.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Finds the userId from event (if given) at different levels as defined in Genric Mapping
+ * @param {*} event input payload to find userId from
+ * @returns either userId or undefined
+ */
+const findUserId = event => {
+  return (
+    event?.userId ||
+    event?.traits?.userId ||
+    event?.context?.traits?.userId ||
+    event?.traits?.id ||
+    event?.context?.traits?.id
+  );
+};
+
+/**
+ * returns userId or anonymousID from event
+ * @param {*} event input to find userId or anonymousId from
+ * @returns userId or anonymousId or undefined
+ */
+const findUserIdOrAnonymousId = event => {
+  return findUserId(event) || event?.anonymousId;
```

Comment:
Can it happen that neither userId nor anonymousId are present?

Reply:
Yes, it can happen and we will be creating a new anonymousId in that case. [Ref](https://github.com/rudderlabs/rudder-transformer/pull/1634/files#:~:text=%3D%20userIdorAnonymousId%20%7C%7C-,generateUUID,-()%3B)